src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -16,18 +15,125 @@
+   * @return either an optional holding the distance topology graph instance or an empty optional
+   */
+  public static Optional<DistanceWeightedGraph> getDistanceTopologyGraph(GridContainer grid) {
+    return getDistanceTopologyGraph(grid.getRawGrid());
+  }
+
+  /**
+   * Returns the topology of the provided {@link RawGridElements} as a {@link
+   * DistanceWeightedGraph}, if they allow the creation of a valid topology graph or an empty
+   * optional otherwise.
+   *
+   * @param rawGridElements raw grids elements as base of the distance weighted topology graph
+   * @return either an optional holding the distance topology graph instance or an empty optional
+   */
+  public static Optional<DistanceWeightedGraph> getDistanceTopologyGraph(
+      RawGridElements rawGridElements) {
+
+    DistanceWeightedGraph graph = new DistanceWeightedGraph();
+
+    try {
+      rawGridElements.getNodes().forEach(graph::addVertex);
+    } catch (NullPointerException ex) {
+      log.error("At least one node entity of provided RawGridElements is null. ", ex);
+      return Optional.empty();
+    }
+
+    try {
+      rawGridElements
+          .getLines()
+          .forEach(line -> addDistanceGraphEdge(graph, line.getNodeA(), line.getNodeB()));
+    } catch (NullPointerException | IllegalArgumentException | UnsupportedOperationException ex) {
+      log.error("Error adding line edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    try {
+      rawGridElements
+          .getSwitches()
+          .forEach(
+              switchInput ->
+                  addDistanceGraphEdge(graph, switchInput.getNodeA(), switchInput.getNodeB()));
+    } catch (NullPointerException | IllegalArgumentException | UnsupportedOperationException ex) {
+      log.error("Error adding switch edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    try {
+      rawGridElements
+          .getTransformer2Ws()
+          .forEach(trafo2w -> addDistanceGraphEdge(graph, trafo2w.getNodeA(), trafo2w.getNodeB()));
+    } catch (NullPointerException | IllegalArgumentException | UnsupportedOperationException ex) {
+      log.error("Error adding 2 winding transformer edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    try {
+      rawGridElements
+          .getTransformer3Ws()
+          .forEach(
+              trafo3w -> {
+                addDistanceGraphEdge(graph, trafo3w.getNodeA(), trafo3w.getNodeB());
+                addDistanceGraphEdge(graph, trafo3w.getNodeA(), trafo3w.getNodeC());
+              });
+    } catch (NullPointerException | IllegalArgumentException | UnsupportedOperationException ex) {
+      log.error("Error adding 3 winding transformer edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    // if we reached this point, we can safely return a valid graph
+    return Optional.of(graph);
+  }
+
+  private static void addDistanceGraphEdge(
```

Comment:
![](https://media.giphy.com/media/l46Cq3dy04lgkRVOE/source.gif)

Huhu! JavaDoc, where are you?